tests/unit/utils/labels/test_geojson_enrichment.py

```
@@ -0,0 +1,624 @@
+        ):
+            enrich_geojson_with_specific_mapping(
+                feature_collection, mock_json_interface, "CLASSIFICATION_JOB", "CLASS_CATEGORY"
+            )
+
+    def test_nonexistent_category_raises_error(self, mock_json_interface):
+        """Test that specifying a non-existent category raises ValueError."""
+        feature_collection = {
+            "type": "FeatureCollection",
+            "features": [
+                {
+                    "type": "Feature",
+                    "geometry": {"type": "Point", "coordinates": [0, 0]},
+                    "properties": {},
+                }
+            ],
+        }
+
+        with pytest.raises(
+            ValueError, match="Category 'NONEXISTENT_CATEGORY' not found in job 'MARKER_JOB'"
+        ):
+            enrich_geojson_with_specific_mapping(
+                feature_collection, mock_json_interface, "MARKER_JOB", "NONEXISTENT_CATEGORY"
+            )
+
+    def test_incompatible_geometry_tool_combination_skipped(self, mock_json_interface):
+        """Test that incompatible geometry-tool combinations are skipped."""
+        feature_collection = {
+            "type": "FeatureCollection",
+            "features": [
+                {
+                    "type": "Feature",
+                    "geometry": {"type": "Point", "coordinates": [0, 0]},
+                    "properties": {},
+                },
+                {
+                    "type": "Feature",
+                    "geometry": {"type": "LineString", "coordinates": [[0, 0], [1, 1]]},
+                    "properties": {},
+                },
+            ],
+        }
+
+        # Try to map to a job that only has polygon tool
+        result = enrich_geojson_with_specific_mapping(
+            feature_collection, mock_json_interface, "POLYGON_JOB", "POLYGON_CATEGORY"
+        )
+
+        # Both features should be skipped as they're incompatible with polygon tool
+        assert len(result["features"]) == 0
```

maybe we should throw a specific error in this case ?